src/include/platform/PlatformManager.h

```diff
@@ -354,9 +355,19 @@ inline void PlatformManager::UnlockChipStack()
     static_cast<ImplClass *>(this)->_UnlockChipStack();
 }
 
-inline void PlatformManager::PostEvent(const ChipDeviceEvent * event)
+inline CHIP_ERROR PlatformManager::PostEvent(const ChipDeviceEvent * event)
```

Comment:
I'd consider the inability to post an event a fairly egregious fault in the system, since there is very little recovery you can do from that situation that doesn't break functionality that was dependent on posting this. Historically on embedded platforms, we'd assert() in that situation so that it's more immediately discoverable and actionable, so that we can actually perhaps alter the size of the event queue, or debug why it become full.

Just emitting a log line feels like it doesn't raise attention enough to this issue.

Reply:
> I'd consider the inability to post an event a fairly egregious fault in the system, since there is very little recovery you can do from that situation that doesn't break functionality that was dependent on posting this.

That's often true, but the case from #9543 is one where it really is good to accept  event-posting failure. Otherwise it becomes trivial to knock out a device with a UDP flood, _or_ there needs to be some other mechanism to prevent that (e.g. have fewer packet buffers than event slots, which practically speaking integrators will not get consistently right).

> Historically on embedded platforms, we'd assert() in that situation so that it's more immediately discoverable and actionable, so that we can actually perhaps alter the size of the event queue, or debug why it become full.

It's always been possible for platforms' `_PostEvent()` to abort on failure — but none of them do, which suggests that leaving it entirely up to platform implementors doesn't work.

> Just emitting a log line feels like it doesn't raise attention enough to this issue.

How about `PostEventOrDie` instead of `PostEventLoggingErrors`?